util/tune/current_settings.rb

```diff
@@ -362,38 +363,43 @@ def puppetdb_command_processing_threads
 def current_settings
-  params = []
-  params << { "puppet_enterprise::master::puppetserver::jruby_max_active_instances" => puppetserver_jruby_max_active_instances }
-  params << { "puppet_enterprise::master::puppetserver::reserved_code_cache" => puppetserver_reserved_code_cache }
-  params << { "puppet_enterprise::profile::console::java_args" => console_java_args }
-  params << { "puppet_enterprise::profile::database::shared_buffers" => database_shared_buffers }
-  params << { "puppet_enterprise::profile::database::autovacuum_max_workers" => database_autovacuum_max_workers }
-  params << { "puppet_enterprise::profile::database::autovacuum_work_mem" => database_autovacuum_work_mem }
-  params << { "puppet_enterprise::profile::database::maintenance_work_mem" => database_maintenance_work_mem }
-  params << { "puppet_enterprise::profile::database::max_connections" => database_max_connections }
-  params << { "puppet_enterprise::profile::database::work_mem" => database_work_mem }
-  params << { "puppet_enterprise::profile::master::java_args" => master_java_args }
-  params << { "puppet_enterprise::profile::orchestrator::java_args" => orchestrator_java_args }
-  params << { "puppet_enterprise::profile::puppetdb::java_args" => puppetdb_java_args }
-  params << { "puppet_enterprise::puppetdb::command_processing_threads" => puppetdb_command_processing_threads }
-
-  json = JSON.pretty_generate params
+  params = {}
+  params["puppet_enterprise::master::puppetserver::jruby_max_active_instances"] = puppetserver_jruby_max_active_instances
+  params["puppet_enterprise::master::puppetserver::reserved_code_cache"] = puppetserver_reserved_code_cache
+  params["puppet_enterprise::profile::console::java_args"] = console_java_args
+  params["puppet_enterprise::profile::database::shared_buffers"] = database_shared_buffers
+  params["puppet_enterprise::profile::database::autovacuum_max_workers"] = database_autovacuum_max_workers
+  params["puppet_enterprise::profile::database::autovacuum_work_mem"] = database_autovacuum_work_mem
+  params["puppet_enterprise::profile::database::maintenance_work_mem"] = database_maintenance_work_mem
+  params["puppet_enterprise::profile::database::max_connections"] = database_max_connections
+  params["puppet_enterprise::profile::database::work_mem"] = database_work_mem
+  params["puppet_enterprise::profile::master::java_args"] = master_java_args
+  params["puppet_enterprise::profile::orchestrator::java_args"] = orchestrator_java_args
+  params["puppet_enterprise::profile::puppetdb::java_args"] = puppetdb_java_args
+  params["puppet_enterprise::puppetdb::command_processing_threads"] = puppetdb_command_processing_threads
+
+  settings_json = JSON.pretty_generate params
 
   # TODO: make this optional?
-  puts json
+  puts settings_json
+
+  # TODO: make the output location optional?
+  FileUtils.mkdir_p DEFAULT_WORKING_DIRECTORY
+  File.write("#{DEFAULT_WORKING_DIRECTORY}/#{DEFAULT_OUTPUT_FILE}", settings_json)
```

Comment:
make sure it will overwrite an existing file (which I think is the right thing to do)

Reply:
I tested this on the master and it does overwrite if the file exists.